core/src/main/scala/org/http4s/multipart/MultipartParser.scala

```diff
@@ -17,47 +17,67 @@
 package org.http4s
 package multipart
 
-import cats._
 import cats.effect.Concurrent
 import cats.syntax.all._
 import fs2.{Chunk, Pipe, Pull, Pure, Stream}
 import fs2.io.file.Files
 import java.nio.file.{Path, StandardOpenOption}
 import org.typelevel.ci.CIString
+import fs2.RaiseThrowable
 
 /** A low-level multipart-parsing pipe.  Most end users will prefer EntityDecoder[Multipart]. */
 object MultipartParser {
-  private[this] val logger = org.log4s.getLogger
-
   private[this] val CRLFBytesN = Array[Byte]('\r', '\n')
   private[this] val DoubleCRLFBytesN = Array[Byte]('\r', '\n', '\r', '\n')
   private[this] val DashDashBytesN = Array[Byte]('-', '-')
   private[this] val BoundaryBytesN: Boundary => Array[Byte] = boundary =>
     boundary.value.getBytes("UTF-8")
   val StartLineBytesN: Boundary => Array[Byte] = BoundaryBytesN.andThen(DashDashBytesN ++ _)
 
+  /** `delimiter` in RFC 2046 */
   private[this] val ExpectedBytesN: Boundary => Array[Byte] =
     BoundaryBytesN.andThen(CRLFBytesN ++ DashDashBytesN ++ _)
   private[this] val dashByte: Byte = '-'.toByte
   private[this] val streamEmpty = Stream.empty
-  private[this] val PullUnit = Pull.pure[Pure, Unit](())
 
   private type SplitStream[F[_]] = Pull[F, Nothing, (Stream[F, Byte], Stream[F, Byte])]
-  private type SplitFileStream[F[_]] =
-    Pull[F, Nothing, (Stream[F, Byte], Stream[F, Byte], Option[Path])]
+
+  private[this] sealed trait Event
```

Comment:
Way back before it was called fs2, I wrote on that was a stream of `Either[Headers, ByteVector]`.  This is better than that, because Left and Right were arbitrary.  That one was public API, and I don't know how many people used it, but it was a nice foundation.